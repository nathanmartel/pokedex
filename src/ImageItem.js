import React, { Component } from 'react';

// Data Object example:
// {
// url:
// "http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg",
// title: "UniWhal",
// description: "A unicorn and a narwhal nuzzling their horns",
// keyword: "narwhal",
// horns: 1
// }


export default class ImageItem extends Component {
    render() {
    return <li>
        <div className="card-image">
            <img src={this.props.creature.url} alt={this.props.creature.title} />
        </div>
        <div className="card-text">
            <h3>{this.props.creature.title}</h3>
            <p>{this.props.creature.description}</p>
            <p>Horns: {this.props.creature.horns}</p>
        </div>
    </li>
    }
}