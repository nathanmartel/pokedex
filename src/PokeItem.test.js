import React from 'react';
import renderer from 'react-test-renderer';
import ImageItem from './ImageItem.js';

const testObject = {
  url:
  "http://3.bp.blogspot.com/_DBYF1AdFaHw/TE-f0cDQ24I/AAAAAAAACZg/l-FdTZ6M7z8/s1600/Unicorn_and_Narwhal_by_dinglehopper.jpg",
  title: "UniWhal",
  description: "A unicorn and a narwhal nuzzling their horns",
  keyword: "narwhal",
  horns: 1
}

it('ImageItem renders a UniWhal', () => {
  const tree = renderer
    .create(<ImageItem creature={testObject} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});