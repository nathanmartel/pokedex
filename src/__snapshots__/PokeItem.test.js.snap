// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokeItem renders a Pidgey 1`] = `
<li>
  <div
    className="card-image"
  >
    <img
      alt="pidgey"
      src="http://assets.pokemon.com/assets/cms2/img/pokedex/full/016.png"
    />
  </div>
  <div
    className="card-text"
  >
    <h2>
      <a
        href="http://www.pokemon.com/us/pokedex/pidgey"
      >
        Pidgey
      </a>
    </h2>
    <p>
      Evolves from 
      NA
    </p>
  </div>
  <div
    className="card-table"
  >
    <table>
      <tr>
        <td
          class="left"
        >
          Attack
        </td>
        <td>
          45
        </td>
      </tr>
      <tr>
        <td
          class="left"
        >
          Defense
        </td>
        <td>
          40
        </td>
      </tr>
      <tr>
        <td
          class="left"
        >
          HP
        </td>
        <td>
          40
        </td>
      </tr>
      <tr>
        <td
          class="left"
        >
          Special Attack
        </td>
        <td>
          35
        </td>
      </tr>
      <tr>
        <td
          class="left"
        >
          Defense
        </td>
        <td>
          35
        </td>
      </tr>
    </table>
  </div>
</li>
`;
